types/enzyme/enzyme-tests.tsx

```diff
@@ -184,6 +184,7 @@ function ShallowWrapperTest() {
         anotherStatelessWrapper = shallowWrapper.find(AnotherStatelessComponent);
         shallowWrapper = shallowWrapper.find({ prop: 'value' });
         elementWrapper = shallowWrapper.find('.selector');
+        anotherComponentWrapper = shallowWrapper.find<AnotherComponentProps>("AnotherComponent");
```

Comment:
it seems kind of strange to pass the props type as a generic here, but then pass a component display name instead. There's no guaranteed unique relationship between `AnotherComponentProps` and `AnotherComponent`'s props type - 10 components might have that type, for example. It seems like the type for `AnotherComponent` is what should be passed in here? (and if you have that, why wouldn't you pass in `AnotherComponent` by constructor instead of by string name?)

Reply:
Yeah I get that. That's the difficult/awkward part. This is mainly to allow people to find components by name without losing the ability to observe props. But I'm having a hard time thinking about how to introduce that in a normal way.

The reason for passing the string name instead of the constructor is that in the past I've seen situations where people had to perform `find`s by name instead of by constructor. I suppose it's possible that Enzyme resolved this quirk in one of its updates. If so, I was unaware; such a thing would make this change unnecessary.

Reply:
It's not an enzyme quirk afaik, it's just a poor structure in the code being tested. In enzyme's gitter, or in an issue, I'd be happy to help you fix that.

The types for passing in a string name should ideally work, but since `find` also supports a string *selector*, this change might not be helpful. In general, passing a string component name is discouraged :-)

Reply:
Just saw this comment, sorry.

Yeah that makes sense. I'll do some more testing with the components that previously had those issues and see if they're still a problem with the latest version. If so I'll open an issue over there. Thanks!